services/arrow-flight/src/main/java/com/dremio/service/flight/impl/GetCatalogsResponseHandler.java

```diff
@@ -0,0 +1,53 @@
+package com.dremio.service.flight.impl;
+
+import org.apache.arrow.flight.FlightProducer;
+import org.apache.arrow.flight.sql.FlightSqlProducer;
+import org.apache.arrow.memory.BufferAllocator;
+import org.apache.arrow.vector.VarCharVector;
+import org.apache.arrow.vector.VectorSchemaRoot;
+import org.apache.arrow.vector.util.Text;
+
+import com.dremio.common.utils.protos.QueryWritableBatch;
+import com.dremio.exec.proto.GeneralRPCProtos;
+import com.dremio.exec.proto.UserProtos;
+import com.dremio.exec.rpc.RpcOutcomeListener;
+import com.dremio.exec.work.protector.UserResponseHandler;
+import com.dremio.exec.work.protector.UserResult;
+
+class GetCatalogsResponseHandler implements UserResponseHandler {
+  private final BufferAllocator allocator;
+  private final FlightProducer.ServerStreamListener listener;
+
+  public GetCatalogsResponseHandler(BufferAllocator allocator, FlightProducer.ServerStreamListener listener) {
+    this.allocator = allocator;
+    this.listener = listener;
+  }
+
+  @Override
+  public void sendData(RpcOutcomeListener<GeneralRPCProtos.Ack> outcomeListener,
+                       QueryWritableBatch result) {
+  }
+
+  @Override
+  public void completed(UserResult result) {
+    UserProtos.GetCatalogsResp catalogsResp = result.unwrap(UserProtos.GetCatalogsResp.class);
+
+    try (VectorSchemaRoot vectorSchemaRoot = VectorSchemaRoot.create(FlightSqlProducer.Schemas.GET_CATALOGS_SCHEMA,
+      allocator)) {
+      listener.start(vectorSchemaRoot);
+
+      vectorSchemaRoot.allocateNew();
```

Comment:
This seems wrong - we're sending data on the completed event?

Reply:
Another implementation of UserResponseHandler:`ResponseSenderHandler`, works just like this and it's used for these catalog functions

Reply:
Yes - but see `RunQueryResponseHandler` as what I think is more representative given they both return result sets.